lib/build.py

```diff
@@ -274,6 +268,215 @@ def build_arg_get(arg):
+
+      # Count the number of stages (i.e., FROM instructions)
+      global image_ct
+      image_ct = sum(1 for i in tree.children_("from_"))
+
+      ml = traverse_parse_tree(tree)
+   else:
+      # Interactive case
+
+      # Generate “fake” SID for build cache. We do this because we can’t compute
+      # an SID, but we still want to make sure that it’s unique enough that
+      # we’re unlikely to run into a collision.
+      fake_sid = uuid.uuid4()
+      out_image.unpack_clear()
+      out_image.copy_unpacked(src_image)
+      bu.cache.worktree_adopt(out_image, src_image.ref.for_path)
+      bu.cache.ready(out_image)
+      bu.cache.branch_nocheckout(src_image.ref, out_image.ref)
+      foo = subprocess.run([ch.CH_BIN + "/ch-run", "--unsafe", "-w",
+                            str(out_image.ref), "--", shell])
+      if (foo.returncode == ch.Ch_Run_Retcode.ERR_CMD.value):
+         # FIXME: Write a better error message?
+         ch.FATAL("Unable to run shell: %s" % shell)
+      ch.VERBOSE("using SID %s" % fake_sid)
+      # FIXME: metadata history stuff? See misc.import_.
+      if (out_image.metadata["history"] == []):
+         out_image.metadata["history"].append({ "empty_layer": False,
+                                                "command":     "ch-image import"})
+      out_image.metadata_save()
+      bu.cache.commit(out_image.unpack_path, fake_sid, "MODIFY interactive", [])
+
+def modify_tree_make(src_img, cmds):
+   """Function that manually constructs a parse tree corresponding to a set of
+      “ch-image modify” commands, as though the commands had been specified in a
+      Dockerfile. Note that because “ch-image modify” simply executes one or
+      more commands inside a container, the only Dockerfile instructions we need
+      to consider are “FROM” and “RUN”. E.g. for the command line
+
+         $ ch-image modify -c 'echo foo' -c 'echo bar' -- foo foo2
+
+      this function produces the following parse tree
+
+         start
+            dockerfile
+               from_
+                  image_ref
+                     IMAGE_REF foo
+               run
+                  run_shell
+                     LINE_CHUNK echo foo
+               run
+                  run_shell
+                     LINE_CHUNK echo bar
+      """
+   # Children of dockerfile tree
+   df_children = []
+   # Metadata attribute. We use this attribute in the “_pretty” method for our
+   # “Tree” class. Constructing a tree without specifying a “Meta” instance that
+   # has been given a “line” value will result in the attribute not being present,
+   # which causes an error when we try to access that attribute. Here we give the
+   # attribute a debug value of -1 to avoid said errors.
+   meta = lark.tree.Meta()
+   meta.line = -1
+   df_children.append(im.Tree(lark.Token('RULE', 'from_'),
+                      [im.Tree(lark.Token('RULE', 'image_ref'),
+                      [lark.Token('IMAGE_REF', str(src_img))], meta)], meta))
```

Comment:
Indent to reflect structure, e.g. something like:

```suggestion
   df_children.append(im.Tree(lark.Token('RULE', 'from_'),
                      [im.Tree(lark.Token('RULE', 'image_ref'),
                        [lark.Token('IMAGE_REF', str(src_img))],
                        meta)
                      ], meta))
```